spec-tests/src/test/java/org/candlepin/spec/environments/EnvironmentSpecTest.java

```diff
@@ -144,6 +144,44 @@ public void shouldFindEnvironmentByName() {
             .returns(env.getId(), EnvironmentDTO::getId);
     }
 
+    @Test
+    public void shouldFindTypedEnvironments() {
+        EnvironmentDTO env1 = ownerClient.owners().createEnv(owner.getKey(), Environments.random()
+            .type(null));
+        EnvironmentDTO env2 = ownerClient.owners().createEnv(owner.getKey(), Environments.random()
+            .type("test_type"));
+        EnvironmentDTO env3 = ownerClient.owners().createEnv(owner.getKey(), Environments.random()
+            .type("another_type"));
+
+        List<EnvironmentDTO> found1 = ownerClient.environments().getEnvironments(null, false);
+        assertThat(found1).contains(env1);
```

Comment:
`.contains` only verifies that the element(s) is contained; it doesn't do verification that other elements *aren't* present.

This should probably be something like `.containsOnly` along with a `.isNotNull()`